Fix Env: Wrap App Name in Quotes

ardelato opened the pull request:
## Description

We need to wrap the `APP_NAME` env var in quotes to avoid issues when generating the final `.env` file. This is similar to what we did for the `META_TWITTER_IMAGE_ALT` env var.'

### CR Notes

This allows us to have a space in the tab title:
![image](https://github.com/user-attachments/assets/5390fc32-b470-4a6a-8777-63c146a1365d)

qa_req 0

ardelato committed:
fix(env): wrap app name in quotes

We need to wrap the APP_NAME env var in quotes to avoid issues when generating
the final .env file. This is similar to what we did for the
META_TWITTER_IMAGE_ALT env var.

djmetzle reviewed (dismissed):
CR :wrench: 

But oof, the whole rest of this file probably ought to be quoted as well?

A note: The curlies are unnecessary when quoted. Its a little cleaner like this:
```sh
APP_NAME="$APP_NAME"
```
(but thats not a blocker)

ardelato committed:
fix(env): wrap environment variables in quotes for consistency

Updated .env.base, .env.docker, .env.template, and docker-compose.yml files to
ensure all environment variables are wrapped in quotes.

djmetzle commented on docker-compose.yml:
```diff
@@ -98,16 +98,16 @@ services:
   restarters_db:
     profiles: ["core", "debug", "discourse"]
     image: mysql:5.7.33
-    container_name: ${MYSQL_CONTAINER_NAME}
```
Careful with these. docker-compose can act a bit weird with templating and escaping.

ardelato replied:
Addressed in [2913b13](https://github.com/iFixit/restarters/pull/25/commits/2913b1315c958dd7611be9b608ab60ad8cc15504)

ardelato committed:
revert(docker-compose): switch back to curly braces for env vars

Instead of using quotes, we will revert back to curly braces for env vars.

mlahargou reviewed (approved):
CR 📱
